### ISSUE TITLE
FileNotFoundError (file exists)

### ISSUE THREAD

**Issue by LordMartron94** [wileyyugioh/zotmoov#21]
This time, a real issue...

I am getting a lot of these:
![image](https://github.com/wileyyugioh/zotmoov/assets/85250199/3f30210a-d8f1-4fad-8c2f-b7daeb0a2bb8)

The file does exist in the path, as is visible here:
![image](https://github.com/wileyyugioh/zotmoov/assets/85250199/830b30b3-8cd1-4655-8111-48be025a7ffd)

Manually copying the pdf to the specified location does work. For some reason, other files do work when moving. (PS, my target is a mapped network drive from my Synology NAS)

**Comment by LordMartron94** [comment]
Btw, I tried forking your repository to fix this issue myself, but for some reason I can't get my fork to work in Zotero... I created a makefile for Windows (I work on Windows) that does the EXACT same as your Unix version. The output is the exact same, and I didn't change any lines of code in my most recent tests, but Zotero still gives the error that it can't run the Startup method in bootstrap even though your release version (exact same source code and exact same build contents) does work without issue (well, except for what I wrote above)... After that it gives the error invalid filename zotmoov.js lol. As I said, didn't change anything in the xpi source code... It's frustrating me as I probably can fix this issue and submit a pull request if I can just get the fork to work lol.

It's really weird and even after hours of research I still can't find the issue.

Makefile Windows:
```makefile
all: get_version.ps1 Makefile.in

-include Makefile.in

# Combine the PowerShell command parts to form the final RELEASE value
RELEASE := $(shell powershell -File "get_version.ps1")

zotmoov.xpi: FORCE
	if exist $@ del /F /Q $@
	"$(ComSpec)" /c "powershell -Command Compress-Archive -Path bootstrap.js, locale, manifest.json, prefs.js, prefs.xhtml, src, lib -DestinationPath temp.zip -Force"
	ren temp.zip $@

zotmoov-%-fx.xpi: zotmoov.xpi
	if exist $@ del /F /Q $@
	ren $< $@

Makefile.in: manifest.json
	echo all: zotmoov-$(RELEASE)-fx.xpi > Makefile.in

FORCE:
```

**Comment by LordMartron94** [comment]
Update: Following the dev environment setup recommended by Zotero: https://www.zotero.org/support/dev/client_coding/plugin_development 

When letting Zotero build it directly from source it does seem to work. However, I did change the id and author for testing purposes. I'll revert that now and see if everything still works. As soon as I get a default working version upon execution directly in Zotero I will see if that same version works when making an xpi.

**Comment by LordMartron94** [comment]
Nope... then I get this again:
![image](https://github.com/wileyyugioh/zotmoov/assets/85250199/6fc00d16-da07-4fcc-b70e-0fd4fad9ce92)

**Comment by LordMartron94** [comment]
I'll go to sleep now. Will await your response, otherwise I will just continue fixing my original issue on the dev environment and once it's fixed I'll submit a pull request so you can add it to the working release.

**Comment by wileyyugioh** [comment]
That is a really weird you can't build it on Windows. .xpi files are essentially zip files with the extension changed. It seems to me like maybe the files in the src folder aren't being copied? Worst case if you can't get it working is to simply select all the files (not the folder) right click > send to > make archive and manually change the file extension.

As for the error in zotmoov.js, it seems that the file_path variable it is getting is not a string, but something else (probably null). The null seems to be from the Zotero API itself. Maybe something to do with it being stored on the server?

**Comment by LordMartron94** [comment]
Yeah, as mentioned, I found a workaround for the build not working; I'll see if I can fix the FileNotFoundError... 

If you want, you can hit me up on Discord btw: mr.hoornasp.learningexpert

**Comment by LordMartron94** [comment]
One step further... 

![image](https://github.com/wileyyugioh/zotmoov/assets/85250199/305cd8c0-1c3d-4595-a067-daf2a7f19556)

The null error was caused, logically, because this specific item:
![image](https://github.com/wileyyugioh/zotmoov/assets/85250199/91798eba-6cb4-4b94-b9a5-0245565b454e)

It has a linked Pubmed entry, which it can't find (and shouldn't link)...

However, the actual pdf is still not moved without throwing an error, so the bug becomes even more difficult now. I think I'll start with cleaning up the several code files (making them SOLID) to make debugging easier.

**Comment by wileyyugioh** [comment]
Are you hitting the maximum path length of windows (260 characters)? I don’t think I have a check for that in the code

**Comment by wileyyugioh** [comment]
The linked pubmed is interesting. I will add a check in the future for attachments without filepaths

**Comment by LordMartron94** [comment]
Nah not hitting the maximum path length (I think) because manually copying the file works as intended.

PS) refactoring this beast is a hell lol. But also fun. Learning a lot.

**Comment by LordMartron94** [comment]
Update: I changed all scripts to classes (which makes it far easier to work with) and got a dynamic script loader to work (the script names are still hardcoded, but it's far easier to add new scripts).

Solid changes are upcoming. Once everything is SOLID and the behavior remains the same, I will submit a pull request.

**Comment by LordMartron94** [comment]
Yeah, I have figured out the issue... My path is too long for specific items (296 characters instead of the Windows limit 260)... However, I am moving items to my NAS, which runs on Linux, so I can manually move them there, as the path limit on Linux is 4096 characters. Now, on to figuring out how to tell IOUtils to disregard the 260-character limit.

**Comment by LordMartron94** [comment]
@wileyyugioh Do you remember if I ever fixed this issue on my end? I reinstalled my laptop and installed Zotero again (I forgot to backup lol) and I am encountering the exact same issue...

**Comment by LordMartron94** [comment]
Nevermind, I found my fixes and pushed them, as well as opened a pull request. (either merge them into the beta/dev branch or wait a couple of days/weeks until I incorporate your custom wildcard behavior)
